OpenSim/Common/Component.h

```diff
@@ -813,7 +811,19 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
         auto it = _connectorsTable.find(name);
 
         if (it != _connectorsTable.end()) {
-            return get_connectors(it->second);
+            // The following allows one to use a Connector immediately after
+            // copying the component;
+            // e.g., myComponent.clone().getConnector("a").getConnecteeName().
+            // Since we use the default copy constructor for Component,
+            // the copied AbstractConnector cannot know its new owner
+            // immediately after copying.
+            if (!it->second->hasOwner()) {
+                // The `this` pointer must be non-const because the Connector
+                // will want to be able to modify the connectee_name property.
+                const_cast<AbstractConnector*>(it->second.get())->setOwner(
+                        const_cast<Self&>(*this));
```

Comment:
I agree, and that was my first plan. But then I ran into the nice code from the README:

```cpp
Ellipsoid bodyGeometry(0.1, 0.5, 0.1);
...
humerusCenter->attachGeometry(bodyGeometry.clone()); <--- no chance to call finalizeFromProperties()
```

This causes the following to be invoked:

```cpp
bodyGeometry.clone()->updConnector<Frame>("frame").setConnecteeName(...);
```

But since we're working with a copy of the original `bodyGeometry`, calling `setConnecteeName()` doesn't work as the copy doesn't know about its parent yet.


These are the options I've thought of so far:
1. Prohibit the concise syntax above and require calling `finalizeFromProperties()`.
2. Cause the same behavior for subcomponents; e.g., have `getComponent()`, `getComponentList()`, etc invoke `setParent()` before returning the subcomponent.
3. When a user calls `Geometry::setFrame()`, or more generally sets up connections via the API, do not store the connectee name at that time; hold onto the given reference. Only fill in the connectee_name property when serializing, etc. This change would have other benefits too (avoid issues with connecting to an object that isn't part of the model yet). However, this is also out of scope of this PR.

Another possibility to